tests/autodiff/generic-jvp.slang

```diff
@@ -115,7 +121,7 @@ extension myfloat3 : IDifferentiable
     [ForwardDifferentiable]
     static Differential dmul<T : __BuiltinRealType>(T a, Differential b)
     {
-        return { __realCast<Real, T>(a) * b.val };
+        return myfloat3(__realCast<Real, T>(a) * b.val);
```

Comment:
Why can't we use initializer list anymore?

Reply:
`{ __realCast<Real, T>(a) * b.val };` is producing incorrect results. `{__realCast<Real, T>(a) * b.val}` is not supposed to be differentiating but it is from what I understand.

This is related to https://github.com/shader-slang/slang/issues/4877

Reply:
But with your change, we should map initializer list to ctor call?

Reply:
> But with your change, we should map initializer list to ctor call?

Yes, this is what will happen once #4854 is merged.
Since the correct behavior is to not differentiate the ctor(?), `generic-jvp.slang` will start failing since it is doing something invalid.

Reply:
In that case we should change the expected result or mark init as differentiable, and keep the initializer list here, right?

Reply:
We can just change the result, that is completely fine
I was just worried about how very confusing that would be to include in the https://github.com/shader-slang/slang/pull/4854 PR rather than as something separate so the change is clearer.